src/text_render.rs

```diff
@@ -225,8 +225,10 @@ impl TextRenderer {
             }
 
             let is_run_visible = |run: &cosmic_text::LayoutRun| {
-                let start_y = (text_area.top + run.line_top) as i32;
-                let end_y = (text_area.top + run.line_top + run.line_height) as i32;
+                let start_y = (text_area.top + (run.line_top * text_area.scale)) as i32;
```

Comment:
Could we put the calculations in named variables to indicate they're physical so we don't accidentally break this later on?

e.g.
```rust
let start_y_physical = text_area.top + run.line_top * text_area.scale;
let end_y_physical = start_y_physical + run.line_height * text_area.scale;
```